GitLab Pipeline Status service test failing

:clock11: **When did the problem start?**
~8 days ago, ~26th of June

<!-- Indicate when the problem started -->

:camera: **Live badge**

<!-- Provide a link to the live badge in plain text and markdown.  -->
https://img.shields.io/gitlab/pipeline/gitlab-org/gitlab
![](https://img.shields.io/gitlab/pipeline/gitlab-org/gitlab)

:wrench: **Is the live badge working?**
Yes
<!-- Indicate whether or not the live badge is working. -->

:link: **CircleCI link**

<!-- Provide a link to the failing test in CircleCI. -->
https://app.circleci.com/pipelines/github/badges/daily-tests/309/workflows/360cfa77-ba5c-422c-b2e6-dbabe6152878/jobs/1206

:beetle: **Stack trace**

```
GitlabPipelineStatus [live] Pipeline status (nonexistent repo) [ GET /this-repo/does-not-exist.json ] - [ GET /this-repo/does-not-exist.json ]
/home/circleci/project/shields/core/service-test-runner/cli.js

AssertionError: message mismatch: expected 'inaccessible' to equal 'repo not found'
    at Function._expectField (core/service-test-runner/icedfrisby-shields.js:69:51)
    at IcedFrisbyNock.<anonymous> (core/service-test-runner/icedfrisby-shields.js:54:26)
    at IcedFrisbyNock.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:954:10)
    at invokeNextHook (node_modules/icedfrisby/lib/icedfrisby.js:1003:24)
    at /home/circleci/project/shields/node_modules/icedfrisby/lib/icedfrisby.js:1017:7
    at new Promise (<anonymous>)
    at IcedFrisbyNock._runHooks (node_modules/icedfrisby/lib/icedfrisby.js:976:12)
    at IcedFrisbyNock.run (node_modules/icedfrisby/lib/icedfrisby.js:1276:20)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Context.<anonymous> (node_modules/icedfrisby/lib/icedfrisby.js:1348:9)
```

:bulb: **Possible solution**
Not sure yet
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->

<!-- Love Shields? Please consider donating $10 to sustain our activities:
👉  https://opencollective.com/shields -->

--- a/services/gitlab/gitlab-pipeline-status.service.js
+++ b/services/gitlab/gitlab-pipeline-status.service.js
@@ -32,6 +32,9 @@ const documentation = `
 <p>
   Your badge should be working fine now.
 </p>
+<p>
+  NB - The badge will display 'inaccessible' if the specified repo was not found on the target Gitlab instance.
+</p>
 `
 
 class GitlabPipelineStatus extends BaseSvgScrapingService {

--- a/services/gitlab/gitlab-pipeline-status.tester.js
+++ b/services/gitlab/gitlab-pipeline-status.tester.js
@@ -20,6 +20,13 @@ t.create('Pipeline status (nonexistent branch)')
     message: 'branch not found',
   })
 
+// Gitlab will redirect users to a sign-in page
+// (which we ultimately see as a 503 error) in the event
+// a nonexistent, or private, repository is specified.
+// Given the additional complexity that would've been required to
+// present users with a more traditional and friendly 'Not Found'
+// error message, we will simply display inaccessible
+// https://github.com/badges/shields/pull/5538
 t.create('Pipeline status (nonexistent repo)')
   .get('/this-repo/does-not-exist/master.json')
   .expectBadge({

--- a/services/gitlab/gitlab-pipeline-status.tester.js
+++ b/services/gitlab/gitlab-pipeline-status.tester.js
@@ -24,7 +24,7 @@ t.create('Pipeline status (nonexistent repo)')
   .get('/this-repo/does-not-exist/master.json')
   .expectBadge({
     label: 'build',
-    message: 'repo not found',
+    message: 'inaccessible',
   })
 
 t.create('Pipeline status (custom gitlab URL)')

